alfalfa_web/server/api-v2.js

```diff
@@ -763,100 +843,115 @@ router.get("/sites/:siteId/download", (req, res) => {
  *               foo: d4e2c041-0389-4933-8aa4-016d80283779
  *               bar: 9e2acb8e-974e-406b-a990-48e9743b01de
  */
-router.get("/aliases", (req, res) => {
+router.get("/aliases", (req, res, next) => {
   api
     .listAliases()
-    .then((data) => res.json(data))
-    .catch(() => res.sendStatus(500));
+    .then((payload) => res.json({ payload }))
+    .catch(next);
 });
 
 /**
  * @openapi
  * /aliases/{alias}:
- *   get:
- *     operationId: Get alias
- *     description: Lookup the site id of an alias
+ *   put:
+ *     operationId: Set alias
+ *     description: Create or update an alias to point to a run id
  *     tags:
  *       - Alias
  *     parameters:
  *       - $ref: '#/components/parameters/Alias'
  *     responses:
- *       200:
- *         description: Alias response
- *         content:
- *           application/json:
- *             schema:
- *               type: string
- *             example:
- *               4b8c6b40-f818-11ec-a8bd-75570e3e3a28
+ *       204:
+ *         description: The alias was set successfully
  *       404:
- *         description: Alias does not exist
+ *         description: Run ID does not exist
  *         content:
  *           application/json:
  *             schema:
  *               $ref: '#/components/schemas/Error'
  */
-router.get("/aliases/:alias", (req, res) => {
+router.put("/aliases/:alias", async (req, res, next) => {
   const { alias } = req.params;
+  const { runId } = req.body;
+
+  const error = validate(
+    { runId },
+    {
+      runId: "required|uuid"
+    }
+  );
+  if (error) return res.status(400).json({ error });
+
+  await api
+    .getRunById(runId)
+    .then((run) => {
+      if (run == null) {
+        return res.status(400).json({
+          error: `Run with ID '${runId}' does not exist`
+        });
+      }
+      api
+        .setAlias(alias, run)
+        .then(() => {
+          return res.sendStatus(204);
+        })
+        .catch(next);
+    })
+    .catch(next);
+});
 
+router.all("/aliases/:alias", (req, res, next) => {
```

Comment:
This does a similar thing as the other pre-processors. The reason this is an `all` instead of `param` is because that would fail on the `PUT` endpoint. Because when you `PUT` an alias it doesn't have to exist yet. So if you ever tried to create an alias you'd get a 404 because it didn't exist yet.